app/controllers/api/skills_controller.rb

```diff
@@ -0,0 +1,99 @@
+class Api::SkillsController < Api::ApiController
+
+  def create
+    user = User.find_or_initialize_by(slack_id: params[:user_id])
+    user.name = params[:user_name]
+    user.save!
+
+    # Try to make the input as flexible as possible:
+    # * match a comma or space separated list of skills, by number or name
+    # e.g. 'Eng,Sales','1,2'
+    # * skills can be preceded by + or - to add or remove the skill
+    # e.g. '+Des,-Sa'; removing a skill you don't have is a no-op
+    # * the strings 'all' or 'none'
+    # * anything else, including mixing known and unknown skills in a list,
+    # or empty text, will make no changes and print help:
+    # "Enter your skills from the following list:
+    # 1-Engineering, 2-Sales, 3-Design, 4-Copywriting"
+    # as well as text indicating the above +,-,all,none
+    # Also, mixing +- and bare numbers will result in the help text -
+    # I figure it's too confusing to accept "1,2,+3"
+
+    text = ""
+    command = params.key?(:command) ? params[:command] : "/skills"
+    begin
+      if params.key?(:text)
+        text=params[:text].strip
+        if text.casecmp?('none')
```

Comment:
I can change it, casecmp? returns a boolean, without the ? returns an integer.  Since we don't care if it's less or greater than 'none', I figured the former was more straightforward.